generated/nitclk/session.py

```diff
@@ -0,0 +1,1156 @@
+# This file was generated
+
+import array
+import ctypes
+import datetime
+import threading
+
+import nitclk._attributes as _attributes
+import nitclk._converters as _converters
+import nitclk._library_singleton as _library_singleton
+import nitclk._visatype as _visatype
+import nitclk.errors as errors
+
+# Used for __repr__ and __str__
+import pprint
+pp = pprint.PrettyPrinter(indent=4)
+
+_session_instance = None
+_session_instance_lock = threading.Lock()
+
+
+# Helper functions for creating ctypes needed for calling into the driver DLL
+def get_ctypes_pointer_for_buffer(value=None, library_type=None, size=None):
+    if isinstance(value, array.array):
+        assert library_type is not None, 'library_type is required for array.array'
+        addr, _ = value.buffer_info()
+        return ctypes.cast(addr, ctypes.POINTER(library_type))
+    elif str(type(value)).find("'numpy.ndarray'") != -1:
+        import numpy
+        return numpy.ctypeslib.as_ctypes(value)
+    elif isinstance(value, list):
+        assert library_type is not None, 'library_type is required for list'
+        return (library_type * len(value))(*value)
+    else:
+        if library_type is not None and size is not None:
+            return (library_type * size)()
+        else:
+            return None
+
+
+class SessionReference(object):
+    '''Properties container for NI-TClk attributes.'''
```

Comment:
That's a strange docstring, doesn't say much. Suggestion:

>Reference to an instrument's session. NI-TClk specific properties can be configured through this.

Reply:
Documentation will be updated in a different PR. To keep them small.